src/analyze/analyze-security.c

```diff
@@ -0,0 +1,2038 @@
+
+                        r = table_update(details_table, cell, TABLE_STRING, buf);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to update cell in table: %m");
+                }
+
+                r = table_print(details_table, stdout);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to output table: %m");
+        }
+
+        exposure = DIV_ROUND_UP(badness_sum * 100U, weight_sum);
+
+        for (i = 0; i < ELEMENTSOF(badness_table); i++)
+                if (exposure >= badness_table[i].exposure)
+                        break;
+
+        assert(i < ELEMENTSOF(badness_table));
+
+        if (details_table) {
+                _cleanup_free_ char *clickable = NULL;
+                const char *name;
+
+                /* If we shall output the details table, also print the brief summary underneath */
+
+                if (info->fragment_path) {
+                        r = terminal_urlify_path(info->fragment_path, info->id, &clickable);
+                        if (r < 0)
+                                return log_oom();
+
+                        name = clickable;
+                } else
+                        name = info->id;
+
+                printf("\n%s %sOverall Exposure Level for %s%s: %s%" PRIu64 ".%" PRIu64 " %s%s %s\n",
+                       special_glyph(ARROW),
+                       ansi_highlight(),
+                       name,
+                       ansi_normal(),
+                       colors_enabled() ? strempty(badness_table[i].color) : "",
+                       exposure / 10, exposure % 10,
+                       badness_table[i].name,
+                       ansi_normal(),
+                       special_glyph(badness_table[i].smiley));
+        }
+
+        fflush(stdout);
+
+        if (overview_table) {
+                char buf[DECIMAL_STR_MAX(uint64_t) + 1 + DECIMAL_STR_MAX(uint64_t) + 1];
+                TableCell *cell;
+
+                r = table_add_cell(overview_table, &cell, TABLE_STRING, info->id);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add cell to table: %m");
+                if (info->fragment_path) {
+                        _cleanup_free_ char *url = NULL;
+
+                        r = file_url_from_path(info->fragment_path, &url);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to generate URL from path: %m");
+
+                        (void) table_set_url(overview_table, cell, url);
+                }
+
+                xsprintf(buf, "%" PRIu64 ".%" PRIu64, exposure / 10, exposure % 10);
+                r = table_add_cell(overview_table, &cell, TABLE_STRING, buf);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add cell to table: %m");
+                (void) table_set_align_percent(overview_table, cell, 100);
+
+                r
```

Comment:
`systemd analyze-security sshd` seems to make UBSan complain that a null pointer passed as argument 1, which is declared to never be null:
```sh
../src/basic/format-table.c:277:16: runtime error: null pointer passed as argument 1, which is declared to never be null
    #0 0x7fd690bca734 in table_data_matches ../src/basic/format-table.c:277
    #1 0x7fd690bcb380 in table_add_cell_full ../src/basic/format-table.c:353
    #2 0x56319aa839dd in table_add_cell ../src/basic/format-table.h:36
    #3 0x56319aa8a68e in assess ../src/analyze/analyze-security.c:1450
    #4 0x56319aa903d9 in analyze_security_one ../src/analyze/analyze-security.c:1915
    #5 0x56319aa9153a in analyze_security ../src/analyze/analyze-security.c:2020
    #6 0x56319aa7d298 in do_security ../src/analyze/analyze.c:1707
    #7 0x7fd690ce0f70 in dispatch_verb ../src/basic/verbs.c:121
    #8 0x56319aa7e29d in main ../src/analyze/analyze.c:1962
    #9 0x7fd68e36911a in __libc_start_main (/lib64/libc.so.6+0x2311a)
    #10 0x56319aa674f9 in _start (/home/vagrant/systemd/build/systemd-analyze+0x4b44f9)

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/basic/format-table.c:277:16 in
```